From 5fdec9592285f5976345fbccb3d07fae1245ab53 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Wed, 2 Aug 2017 15:41:22 -0700
Subject: [PATCH] mongo: Add using std::string

This is needed with latest clang
Fixes
src/mongo/db/dbwebserver.cpp:206:23: error: use of undeclared identifier 'string'; did you mean 'String'?
        static vector<string> commands;
                      ^~~~~~
                      String

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 src/mongo/db/dbwebserver.cpp             | 1 +
 src/mongo/db/matcher/expression_leaf.cpp | 2 ++
 src/mongo/db/repl/master_slave.cpp       | 1 +
 src/mongo/util/net/miniwebserver.cpp     | 1 +
 4 files changed, 5 insertions(+)

diff --git a/src/mongo/db/dbwebserver.cpp b/src/mongo/db/dbwebserver.cpp
index b202e0500b..0255613524 100644
--- a/src/mongo/db/dbwebserver.cpp
+++ b/src/mongo/db/dbwebserver.cpp
@@ -67,6 +67,7 @@ namespace mongo {
 using std::map;
 using std::stringstream;
 using std::vector;
+using std::string;
 
 using namespace html;
 
diff --git a/src/mongo/db/matcher/expression_leaf.cpp b/src/mongo/db/matcher/expression_leaf.cpp
index d562bff141..4d931985e2 100644
--- a/src/mongo/db/matcher/expression_leaf.cpp
+++ b/src/mongo/db/matcher/expression_leaf.cpp
@@ -44,6 +44,8 @@
 #include "mongo/stdx/memory.h"
 #include "mongo/util/mongoutils/str.h"
 
+using std::string;
+
 namespace mongo {
 
 Status LeafMatchExpression::setPath(StringData path) {
diff --git a/src/mongo/db/repl/master_slave.cpp b/src/mongo/db/repl/master_slave.cpp
index 05faad1259..ea25d7151b 100644
--- a/src/mongo/db/repl/master_slave.cpp
+++ b/src/mongo/db/repl/master_slave.cpp
@@ -78,6 +78,7 @@ using std::endl;
 using std::max;
 using std::min;
 using std::set;
+using std::string;
 using std::stringstream;
 using std::unique_ptr;
 using std::vector;
diff --git a/src/mongo/util/net/miniwebserver.cpp b/src/mongo/util/net/miniwebserver.cpp
index 5f4165d42f..239720c349 100644
--- a/src/mongo/util/net/miniwebserver.cpp
+++ b/src/mongo/util/net/miniwebserver.cpp
@@ -46,6 +46,7 @@ namespace mongo {
 
 using std::shared_ptr;
 using std::stringstream;
+using std::string;
 using std::vector;
 
 MiniWebServer::MiniWebServer(const string& name, const string& ip, int port, ServiceContext* ctx)
-- 
2.13.3

